lightning/src/ln/channel.rs

```diff
@@ -3771,6 +3991,109 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
+		&self, logger: &L
+	) -> Result<Signature, ChannelError>
+	where
+		SP::Target: SignerProvider,
+		L::Target: Logger
+	{
+		let counterparty_keys = self.build_remote_transaction_keys();
+		let counterparty_initial_commitment_tx = self.build_commitment_transaction(
+			self.cur_counterparty_commitment_transaction_number, &counterparty_keys, false, false, logger).tx;
+		match self.holder_signer {
+			// TODO (taproot|arik): move match into calling method for Taproot
+			ChannelSignerType::Ecdsa(ref ecdsa) => {
+				ecdsa.sign_counterparty_commitment(&counterparty_initial_commitment_tx, Vec::new(), Vec::new(), &self.secp_ctx)
+					.map(|(signature, _)| signature)
+					.map_err(|_| ChannelError::Close(
+						(
+							"Failed to get signatures for new commitment_signed".to_owned(),
+							ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
+						)))
+			},
+			// TODO (taproot|arik)
+			#[cfg(taproot)]
+			_ => todo!(),
+		}
+	}
+
+	fn get_initial_commitment_signed<L: Deref>(
+		&mut self, transaction: &ConstructedTransaction, logger: &L
+	) -> Result<msgs::CommitmentSigned, ChannelError>
+	where
+		SP::Target: SignerProvider,
+		L::Target: Logger
+	{
+		if !matches!(
+			self.channel_state, ChannelState::NegotiatingFunding(flags)
+			if flags == (NegotiatingFundingFlags::OUR_INIT_SENT | NegotiatingFundingFlags::THEIR_INIT_SENT)) {
+			panic!("Tried to get a funding_created messsage at a time other than immediately after initial handshake completion (or tried to get funding_created twice)");
+		}
+		if self.commitment_secrets.get_min_seen_secret() != (1 << 48) ||
+				self.cur_counterparty_commitment_transaction_number != INITIAL_COMMITMENT_NUMBER ||
+				self.holder_commitment_point.transaction_number() != INITIAL_COMMITMENT_NUMBER {
+			panic!("Should not have advanced channel commitment tx numbers prior to initial commitment_signed");
+		}
+
+		let funding_redeemscript = self.get_funding_redeemscript().to_p2wsh();
+		let funding_outpoint_index = transaction.outputs().enumerate().find_map(
+			|(idx, output)| {
+				if output.tx_out().script_pubkey == funding_redeemscript { Some(idx as u16) } else { None }
+			}).expect("funding transaction contains funding output");
+		let funding_txo = OutPoint { txid: transaction.txid(), index: funding_outpoint_index };
+		self.channel_transaction_parameters.funding_outpoint = Some(funding_txo);
+		self.holder_signer.as_mut().provide_channel_parameters(&self.channel_transaction_parameters);
+
+		let signature = match self.get_initial_counterparty_commitment_signature(logger) {
+			Ok(res) => res,
+			Err(e) => {
+				log_error!(logger, "Got bad signatures: {:?}!", e);
+				self.channel_transaction_parameters.funding_outpoint = None;
+				return Err(e);
+			}
+		};
+
+		if self.signer_pending_funding {
+			log_trace!(logger, "Counterparty commitment signature ready for funding_created message: clearing signer_pending_funding");
+			self.signer_pending_funding = false;
+		}
```

Comment:
As discussed offline, this can be dropped from this PR and re-added for async signing.